### ISSUE TITLE
Caching bigger downloads on the host machine would speed up rebuild times for the vm

### ISSUE THREAD

**Issue by lo5an** [Islandora-Labs/islandora_vagrant#21]
Saving bigger downloads in a cache in the mounted /vagrant folder would speed up subsequent rebuild times.

**Comment by ruebot** [comment]
Thoughts on adding the `downloads` folder to the repo? This would close the ticket, but it would also make the repo bigger, and we'd be distributing a bunch of binaries.

**Comment by lo5an** [comment]
I like the idea of the script downloading stuff, but then caching it between machine rebuilds.  I don't think we want to keep our own copy of the binaries.

**Comment by ruebot** [comment]
Welp, it looks like we can't even do it!

```
$ git push origin issue-21 
Counting objects: 14, done.
Delta compression using up to 4 threads.
Compressing objects: 100% (12/12), done.
Writing objects: 100% (12/12), 345.71 MiB | 1.20 MiB/s, done.
Total 12 (delta 1), reused 0 (delta 0)
remote: warning: File downloads/fits-0.8.4_0.zip is 70.63 MB; this is larger than GitHub's recommended maximum file size of 50 MB
remote: error: GH001: Large files detected.
remote: error: Trace: 9cd5e4b75ac0b6a2810a79d8c84faadd
remote: error: See http://git.io/iEPt8g for more information.
remote: error: File downloads/fcrepo-installer-3.8.0.jar is 145.32 MB; this exceeds GitHub's file size limit of 100 MB
remote: error: File downloads/solr-4.2.0.tgz is 111.14 MB; this exceeds GitHub's file size limit of 100 MB
To git@github.com:Islandora-Labs/islandora_vagrant.git
 ! [remote rejected] issue-21 -> issue-21 (pre-receive hook declined)
error: failed to push some refs to 'git@github.com:Islandora-Labs/islandora_vagrant.git'
```

So, I guess we're good to go with b3ec9b455930a53b4d39039dd7f1b5a632ffd52c closing the ticket. If that's the case, I'll let @lo5an decide since he created the issue.

**Comment by whikloj** [comment]
There is also the possibility of:
1) Replace _apt-get_ with **apt-get -d** to download source into the downloads directory. 
2) Use **dpkg -i** to install the packages from the downloads directory.
This would extend the download once, install many idea. Just not sure how it would handle dependencies.

**Comment by ruebot** [comment]
...and we should probably add `downloads` to `.gitignore`?

**Comment by ruebot** [comment]
@whikloj++ #28

**Comment by lo5an** [comment]
Maybe a vagrant plugin like https://github.com/fgrehm/vagrant-cachier would be better for caching system packages? 

Updated to expand:

I've only been using vagrant-cachier for a week or so, but it's been working well so far, and I like that it handles more than just .debs. It seems to be a reasonably active project, and it's nice and modular, so we could recommend that people use it without making it a dependency. I've got the vagrant-cachier config in a Vagrantfile in my lookup path so that it applies to all of my vagrant vms.
